### PR TITLE
fix(node-dev): fix the custom node build command

### LINKED ISSUE #4133: Unable to build custom node with n8n-node-dev
**Describe the bug**
Following instructions to create a custom node and include it into the ~/.n8n/custom from [https://github.com/n8n-io/n8n/blob/b22ff1f5c17ecfcafd7bb8739b311fd0b88d37e8/packages/node-dev/README.md](readme.md)

I get the following: 

```
**@**:~/projects/n8n$ n8n-node-dev build

Build credentials and nodes
=========================

GOT ERROR: "Unexpected token } in JSON at position 85"
====================================
SyntaxError: Unexpected token } in JSON at position 85
    at JSON.parse (<anonymous>)
    at createCustomTsconfig (**/npm/node_modules/n8n-node-dev/dist/src/Build.js:17:27)
    at async buildFiles (**/npm/node_modules/n8n-node-dev/dist/src/Build.js:41:26)
    at async Build.run (**/npm/node_modules/n8n-node-dev/dist/commands/build.js:20:37)
    at async Build._run (**/npm/node_modules/n8n-node-dev/node_modules/@oclif/command/lib/command.js:43:20)
    at async Config.runCommand (**/npm/node_modules/n8n-node-dev/node_modules/@oclif/config/lib/config.js:173:24)
    at async Main.run (**/npm/node_modules/n8n-node-dev/node_modules/@oclif/command/lib/main.js:27:9)
    at async Main._run (**/npm/node_modules/n8n-node-dev/node_modules/@oclif/command/lib/command.js:43:20)

```
**To Reproduce**
In an empty folder run :

` n8n-node-dev new`
` n8n-node-dev build`

**Expected behavior**
Build is successful 

**Environment (please complete the following information):**

- OS: [e.g. Ubuntu Linux 20.04.5 LTS (Win10 WLS2)]
- n8n Version [n8n@0.194.0]
- Node.js Version [16.17.0]
- Database system [SQLite]
- Operation mode [ `own`]

### LINKED ISSUE #4142: Unable to build custom nodes from n8n-nodes-starter due to eslint 404
**Describe the bug**
When using npm link to connect the n8n global install and the n8n-nodes-start project, you receive a 404. The workflow is described in this stream from n8n: https://www.youtube.com/watch?v=OI6zHJ56eW0 from roughly 4-9 minutes. It looks like this was brought about in PR 4050 (https://github.com/n8n-io/n8n/pull/4050)

npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/@n8n_io%2feslint-config - Not found
npm ERR! 404 
npm ERR! 404  '@n8n_io/eslint-config@latest' is not in this registry.

**To Reproduce**

1. Clone the n8n-nodes-starter project
2. Install n8n globally (npm install n8n -g)
3. From the starter project directory, run (npm link)
4. From your global install directory, ie /usr/local/lib/node_modules, go to the n8n subdirectory and link your starter project (npm link n8n-nodes-<yourprojectname>

**Expected behavior**
The link appears to be referencing a package which has not been published (@n8n_io/eslint-config@latest). On the n8n project page (https://www.npmjs.com/package/n8n) the eslint project is listed, but it directs to a 404. 

**Environment (please complete the following information):**

- OS: MacOS 12.5.1
- n8n 0.194.0
- Node.js Version 18.9.0
- Database system SQLite
- Operation mode own

### PR DESCRIPTION
**Pull request by netroy**
also
* wait for the build command to finish before logging the success message
* use module relative paths for looking up `tsconfig` and `tsc` using `require.resolve`

fixes #4133

### CODE REVIEW

File: packages/node-dev/src/Build.ts [hunk 1]
```diff
@@ -110,8 +94,10 @@ export async function buildFiles(options?: IBuildOptions): Promise<string> {
 
 		// Make sure that the child process gets also always terminated
 		// when the main process does
-		process.on('exit', () => {
-			buildProcess.kill();
+		process.on('exit', () => buildProcess.kill());
+
+		await new Promise<void>((resolve) => {
+			buildProcess.on('exit', () => resolve());
```
Comment (ivov) [inline packages/node-dev/src/Build.ts:100]:
```suggestion
			buildProcess.on('exit', resolve);
```

File: packages/node-dev/src/Build.ts [hunk 2]
```diff
@@ -66,24 +50,25 @@ export async function createCustomTsconfig() {
...
-	let typescriptPath;
-
-	// Check for OS to designate correct tsc path
-	if (process.platform === 'win32') {
-		typescriptPath = '../../node_modules/TypeScript/lib/tsc';
-	} else {
-		typescriptPath = '../../node_modules/.bin/tsc';
-	}
-	const tscPath = join(__dirname, typescriptPath);
+	options = options ?? {};
```
Comment (ivov) [inline packages/node-dev/src/Build.ts:53]: Or as default arg.

**Review by ivov — COMMENTED** [review]
Confirmed working.

1. Unrelated but could you please fix the typo? `The nodes got build` → `The nodes got built`

2. Sadly `tsc` does not have a `--silent` flag and redirecting to `/dev/null` is not cross-platform. Any quick way you can think of to suppress these module-not-found errors? Else disregard.

```
Build credentials and nodes
=========================
MyNode.node.ts(1,35): error TS2307: Cannot find module 'n8n-core' or its corresponding type declarations.
MyNode.node.ts(2,69): error TS2307: Cannot find module 'n8n-workflow' or its corresponding type declarations.
The nodes got build and saved into the following folder:
/Users/ivov/.n8n/custom
```

**Comment by netroy** [comment]
>     1. Unrelated but could you please fix the typo? `The nodes got build` → `The nodes got built`
:heavy_check_mark: 
 
>     2. Sadly `tsc` does not have a `--silent` flag and redirecting to `/dev/null` is not cross-platform. Any quick way you can think of to suppress these module-not-found errors? Else disregard.
This would be possible but not easily in this code. I plan to switch this code over from `child_process` to `createProgram` *from the `typescript` package). But that'll be another PR.

**Review by ivov — APPROVED** [review]

**Comment by janober** [comment]
Got released with `n8n@0.195.0`

**Pull request merged**